src/screens/create-wallet/create-wallet.screen.tsx

```diff
@@ -18,82 +29,116 @@ import {
   Label,
   SignnalingInput,
 } from './create-wallet.screen.styles';
+import { claimTypes, getClaimName } from 'utils';
+import { FormValidator } from 'utils/FormValidator';
 
 export const CreateWalletScreen = observer(() => {
   const { safeService, walletService } = useServices();
-  const { safeStore } = useStores();
+  const { safeStore, accountStore } = useStores();
   let history = useHistory();
 
   const [walletName, setWalletName] = useState('');
   const [walletDescription, setWalletDescription] = useState('');
+
   const [walletBeneficiary, setWalletBeneficiary] = useState(safeService.getDefaultConfig()?.beneficiary);
   const [signalingPeriod, setSignalingPeriod] = useState(300);
+
   const [claimType, setClaimType] = useState(0);
   const [DdayTime, setDdayTime] = useState(0);
+
   const [date, setDate] = useState(null);
+  const [isTopupToggleChecked, setIsTopupToggleChecked] = useState(false);
+
+  const [topupValue, setTopupValue] = useState<any>('');
+
+  const [selectWallet, setSelectWallet] = useState([]);
+  const [options, setOptions] = useState(selectWallet);
+
+  const [seedPhrase, setSeedPhrase] = useState<any>('');
+  const [balanceLoader, setBalanceLoadder] = useState(false);
   const [errorMessage, setErrorMessage] = useState('');
 
+  const [validator, setValidator] = useState(false);
+
+  const getAllWalletsAndFetchBalance = async () => {
+    const wallets = await accountStore.safientUser?.safes.map((safes) => ({
+      label: safes.safeName,
+      value: safes.safeId,
+    }));
+
+    setSelectWallet(wallets);
+    setBalanceLoadder(true);
+    safeStore.setRole('creator');
+    const safeData = await safeService.recover(options, 'creator');
```

Comment:
Separate the wallet balance fetch functionality and call it after the wallet is selected.